processor/deltatocumulativeprocessor/internal/metrics/data.go

```diff
@@ -0,0 +1,56 @@
+package metrics
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/data"
+)
+
+type Data[D data.Point[D]] interface {
+	At(i int) D
+	Len() int
+	Ident() Ident
+}
+
+type Sum Metric
+
+func (s Sum) At(i int) data.Number {
+	dp := Metric(s).Sum().DataPoints().At(i)
```

Comment:
Is casting to `Metric` necessary here?

Reply:
yes, as this as `Sum` is a different type it doesn't have the methods of `Metric`.